docs/content/DimensionSpecs.md

```diff
@@ -154,3 +154,198 @@ Example for the `__time` dimension:
   "function" : "function(t) { return 'Second ' + Math.floor((t % 60000) / 1000); }"
 }
 ```
+
+### Namespaced extraction function (EXPERIMENTAL)
+A namespaced extraction function is a simple key-value mapping where the key-value mappings are unique to a particular namespace.
+This can be used for re-naming values in a datasource for certain queries. To use a particular namespace, simply set the appropriate namespace for the dimExtractionFn
+```json
+    "dimExtractionFn" : {
+      "type":"namespace",
+      "namespace":"some_namespace",
+      "missingValue":"NA",
+      "uniqueRenames":false
+    }
+```
+
+A property of `missingValue` can be specified at query time to hint how to handle missing values. Setting `missingValue` to `""` or `null` or omitting the property will cause the dimension's original value to be used.
+ 
+A property of `uniqueRenames` specifies if optimizations can be used which assume there is no combining of multiple names into one. For example: If ABC123 always maps to SomeCompany, that can be optimized since it is a unique rename. But if both ABC123 and DEF456 BOTH map to SomeCompany, then that is NOT a unique rename.
+ 
+Namespace updates can be set through adding a child in utf8 json form to the zookeeper path at`druid.zk.paths.namespacePath`. It is required that the namespace under the ZK path be equal to the namespace in the json. The json is of the following format:
+
+#### URI namespace update
+The remapping values for each namespace can be specified by a namespace update json object as per
+```json
+{
+  "type":"uri",
+  "namespace":"some_namespace",
+  "uri": "s3://bucket/some/key/prefix/",
+  "parseSpec":{
+    "type":"csv",
+    "columns":["key","value"]
+  },
+  "updateMs":0,
+  "versionRegex": "renames-[0-9]*\\.gz"
+}
+```
+
+The `updateMs` value specifies the period in milliseconds between checks for updates. If the source of the namespace is capable of providing a timestamp, the namespace will only be updated if it has changed since the prior tick of `updateMs`. A value of 0, an absent parameter, or `null` all mean populate once and do not attempt to update. Whenever an update occurs, the updating system will look for a file with the most recent timestamp and assume that one with the most recent data.
```

Comment:
we typically specify polling and refresh times as ISO periods to make it easier to configure, e.g. "PT5M" to refresh every 5 minutes. Something like `pollPeriod` or `updatePeriod` or would be more consistent with existing configuration names.